src/main/java/uk/gov/hmcts/reform/sscs/util/SscsOcrDataUtil.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.reform.sscs.util;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Map;
+
+public class SscsOcrDataUtil {
+
+    private static final String YES = "Yes";
+    private static final String NO = "No";
+
+    private SscsOcrDataUtil() {
+
+    }
+
+    public static Boolean hasPerson(Map<String, Object> pairs, String person) {
+
+        return findBooleanExists(getField(pairs,person + "_title"), getField(pairs,person + "_first_name"),
+            getField(pairs,person + "_last_name"), getField(pairs,person + "_address_line1"), getField(pairs,person + "_address_line2"),
+            getField(pairs,person + "_address_line3"), getField(pairs,person + "_address_line4"), getField(pairs,person + "_postcode"),
+            getField(pairs,person + "_date_of_birth"), getField(pairs,person + "_ni_number"),  getField(pairs,person + "_company"));
+    }
+
+    public static boolean findBooleanExists(String... values) {
+        for (String v : values) {
+            if (v != null) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static String getField(Map<String, Object> pairs, String field) {
+        return pairs.get(field) != null ? pairs.get(field).toString() : null;
+    }
+
+    public static boolean doBooleansContradict(Map<String, Object> pairs, List<String> errors, String value1, String value2) {
+        if (pairs.containsKey(value1) && pairs.containsKey(value2)) {
+            if ((boolean) pairs.get(value1) == (boolean) pairs.get(value2)) {
+                errors.add(value1 + " and " + value2 + " have contradicting values");
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static boolean areBooleansValid(Map<String, Object> pairs, List<String> errors, String... values) {
+        for (String value : values) {
+            if (pairs.get(value) == null) {
+                return false;
+            } else if (!(pairs.get(value) instanceof Boolean)) {
+                errors.add(value + " does not contain a valid boolean value. Needs to be true or false");
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
after all reviewing this looks repetitive now. But there are some delicate differences. why not reducing this into more _recent_ api?
just thinking out of the box but perhaps `Stream.of(values).map(pairs::get)` would work out?

Reply:
Not sure this would bring much benefit as we would still need to break the method out into separate blocks.